imap_processing/ena_maps/utils/spatial_utils.py

```diff
@@ -0,0 +1,270 @@
+"""IMAP utils for spatial binning and az/el grid creation."""
+
+from __future__ import annotations
+
+import typing
+
+import numpy as np
+from numpy.typing import NDArray
+
+
+def build_spatial_bins(
+    az_spacing: float = 0.5,
+    el_spacing: float = 0.5,
+) -> tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    """
+    Build spatial bin boundaries for azimuth and elevation (input/output angles in deg).
+
+    Parameters
+    ----------
+    az_spacing : float, optional
+        The azimuth bin spacing in degrees (default is 0.5 degrees).
+    el_spacing : float, optional
+        The elevation bin spacing in degrees (default is 0.5 degrees).
+
+    Returns
+    -------
+    az_bin_edges : np.ndarray
+        Array of azimuth bin boundary values in degrees.
+    el_bin_edges : np.ndarray
+        Array of elevation bin boundary values in degrees.
+    az_bin_midpoints : np.ndarray
+        Array of azimuth bin midpoint values in degrees.
+    el_bin_midpoints : np.ndarray
+        Array of elevation bin midpoint values in degrees.
+    """
+    # Azimuth bins from 0 to 360 degrees.
+    az_bin_edges = np.arange(0, 360 + az_spacing, az_spacing)
+    az_bin_midpoints = az_bin_edges[:-1] + az_spacing / 2  # Midpoints between edges
+
+    # Elevation bins from -90 to 90 degrees.
+    el_bin_edges = np.arange(-90, 90 + el_spacing, el_spacing)
+    el_bin_midpoints = el_bin_edges[:-1] + el_spacing / 2  # Midpoints between edges
+
+    return az_bin_edges, el_bin_edges, az_bin_midpoints, el_bin_midpoints
```

Comment:
Do we want/need the midpoints? The edges should be good enough to constrain the problem and then let other functions get the midpoints if they need them later. Returning a lot of things from a function can be confusing.

Also, keep in mind the az/el ordering throughout, later you have `el` first.

Reply:
The midpoints are the coordinates of the rectangular 'pixel' on the sky. We definitely need them for constructing the grid later in the AzEl class, and for building the map CDF. Also, some of Laura's L1 code uses them. 

I think it's probably more confusing to need to do something like:

`el_bin_midpoints = el_bin_edges[:-1] + el_spacing / 2  # Midpoints between edges`

in a lot of different locations.

Reply:
Alternatively, do we need the "edges" then, or could they be inferred from the spacing later when they are needed? What I'm getting at is that they are somewhat redundant with one another and it is often good to have one "source of truth" that you can then calculate the other from instead of possibly getting them out of sync at some later point in time.

Reply:
NOTE: We discussed this in person